warehouse/legacy/api/xmlrpc/views.py

```diff
@@ -62,6 +64,28 @@ def wrapped(context, request):
     return decorator
 
 
+def ratelimit():
+    def decorator(f):
+        def wrapped(context, request):
+            ratelimiter = request.find_service(
+                IRateLimiter, name="xmlrpc.client", context=None
+            )
+            if not ratelimiter.test(request.remote_addr):
+                raise XMLRPCWrappedError(
+                    HTTPTooManyRequests(
+                        "The action could not be performed because there were too "
+                        "many requests by the client. Limit resets in "
+                        f"{ratelimiter.resets_in(request.remote_addr)}"
```

Comment:
If I remember correctly, this is when the limit *might* reset. A `RateLimiter` can contain multiple identifiers, and thus this value is meant to be advisory only. It also might be ``None`` if there is a problem talking to redis for some reason.